sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Security/Ucfg/UniversalControlFlowGraphBuilder.cs

```diff
@@ -0,0 +1,331 @@
+            }
+
+            private Expression BuildIdentifier(IdentifierNameSyntax identifier, BasicBlock block)
+            {
+                var identifierSymbol = GetSymbol(identifier);
+                if (identifierSymbol is IPropertySymbol property)
+                {
+                    var tempVariable = GetTempVariable();
+                    block.Instructions.Add(new Instruction
+                    {
+                        MethodId = MethodIdProvider.Create(property.GetMethod),
+                        Variable = tempVariable,
+                    });
+                    return CreateVariable(tempVariable);
+                }
+                else
+                {
+                    return CreateExpression(identifier);
+                }
+            }
+
+            private void BuildVariableDeclarator(VariableDeclaratorSyntax variableDeclarator, BasicBlock block)
+            {
+                if (variableDeclarator.Initializer == null)
+                {
+                    return;
+                }
+
+                var left = semanticModel.GetDeclaredSymbol(variableDeclarator);
+                if (IsLocalVarOrParameterOfTypeString(left))
+                {
+                    var instruction = new Instruction
+                    {
+                        Location = GetLocation(variableDeclarator),
+                        Variable = left.Name,
+                        MethodId = KnownMethodId.Assignment,
+                        Args = { BuildInstruction(variableDeclarator.Initializer.Value, block) },
+                    };
+                    block.Instructions.Add(instruction);
+                }
+            }
+
+            private Expression BuildBinaryExpression(BinaryExpressionSyntax binaryExpression, BasicBlock block)
+            {
+                var instruction = new Instruction
+                {
+                    Location = GetLocation(binaryExpression),
+                    MethodId = KnownMethodId.Concatenation,
+                    Args = { BuildInstruction(binaryExpression.Right, block), BuildInstruction(binaryExpression.Left, block) },
+                    Variable = GetTempVariable(),
+                };
+                block.Instructions.Add(instruction);
+                return CreateVariable(instruction.Variable);
+            }
+
+            public void BuildReturn(ReturnStatementSyntax returnStatement, BasicBlock block)
+            {
+                block.Ret = new Return
+                {
+                    Location = GetLocation(returnStatement),
+                    ReturnedExpression = returnStatement.Expression != null ? BuildInstruction(returnStatement.Expression, block) : CreateConstant(),
+                };
+            }
+
+            private Expression BuildInvocation(InvocationExpressionSyntax invocation, BasicBlock block)
+            {
+                var methodSymbol = GetSymbol(invocation) as IMethodSymbol;
+                if (methodSymbol == null)
```

Comment:
What input would produce null here?

Reply:
When the `methodSymbol` is null we will not generate instruction on invocation statement. However, when the method is invoked as an argument of another expression:
`Foo(Bar(x));` we need to return "something".